envoy/auth_server/config/config_test.go

```diff
@@ -0,0 +1,47 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_LoadAuthServerConfigFromYAML(t *testing.T) {
+	tests := []struct {
+		name     string
+		filePath string
+		want     AuthServerConfig
+		wantErr  bool
+	}{
+		{
+			name:     "should load valid auth server config without error",
+			filePath: "../../../config/examples/config.shannon_example.yaml",
```

Comment:
WHy is the test loading `config.shannon_example.yaml` for the `auth server config`?

Makes me worry about the validation that's happening now.

Reply:
Noted, but is `shannon_example.yaml` a valid `auth server config` or just a `valid config yaml file` that we can load?

Is this testing loading or validation? The `name` is what confuses me.

Reply:
> Noted, but is `shannon_example.yaml` a valid `auth server config` or just a `valid config yaml file` that we can load?

It's both; the `config.shannon_example.yaml` file contains the `auth_server` config. My intention is not to make too many configuration files.

 ie. user wants to configure something in PATH -> go to and modify `.config.yaml`. IMO that makes sense vs having more config YAML files.

> Is this testing loading or validation? The `name` is what confuses me.

In this case it's testing loading. The error test cases below test the validation.